tests/serverpod_test_server/test_e2e/http_codes_test.dart

```diff
@@ -0,0 +1,73 @@
+import 'dart:convert';
+
+import 'package:http/http.dart' as http;
+import 'package:serverpod_test_server/test_util/config.dart';
+import 'package:test/expect.dart';
+import 'package:test/scaffolding.dart';
+
+void main() {
+  group('Given an HTTP client invoking a serverpod endpoint method', () {
+    setUpAll(() async {});
+
+    tearDownAll(() async {});
+
+    test(
+        'when calling an endpoint method with correct parameters '
+        'then it should respond with 200 ok', () async {
+      var response = await http.post(
+        Uri.parse('${serverUrl}simple'),
+        body: jsonEncode({
+          'method': 'hello',
+          'name': 'Starbase Alpha',
+        }),
+      );
+
+      expect(response.statusCode, 200);
+      expect(response.body, contains('Hello Starbase Alpha'));
+    });
+
+    test(
+        'when calling an endpoint method with missing parameters '
+        'then it should respond with 400 bad request', () async {
+      var response = await http.post(
+        Uri.parse('${serverUrl}simple'),
+        body: jsonEncode({
+          'method': 'hello',
+        }),
+      );
+
+      expect(response.statusCode, 400);
+      expect(response.body, contains('Missing required query parameter: name'));
+    });
+
+    test(
+        'when calling an endpoint method with too many parameters '
+        'then it should respond with 400 bad request', () async {
+      var response = await http.post(
+        Uri.parse('${serverUrl}simple'),
+        body: jsonEncode({
+          'method': 'hello',
+          'name': 'Starbase Alpha',
+          'extra': 'spurious value',
+        }),
+      );
+
+      expect(response.statusCode, 400);
+    }, skip: 'desired behavior unclear');
+
+    test(
+        'when calling an endpoint method with misspelled method name '
+        'then it should respond with 400 bad request', () async {
+      var response = await http.post(
+        Uri.parse('${serverUrl}simpleMistake'),
```

Comment:
Suggestion: To make this test more robust you could generate the method name with a UUID.

"misspelled" or "non-existing" does not really matter for this case. To make this test resistant towards someone actually introducing an endpoint called "simpleMistake" we could just as well generate a UUID for the method name and prove the same point.